aptos/api/aux-ts/src/stake/client.ts

```diff
@@ -154,28 +232,146 @@ export class StakePoolClient {
    */
   async calcPendingUserReward({
     userAddress,
-    lastUserPositionState,
+    userPosition,
+    currentTimeUs,
+    lastPoolState,
   }: {
     userAddress: Types.Address;
-    lastUserPositionState?: UserPosition;
+    userPosition?: UserPosition;
+    currentTimeUs?: BN;
+    lastPoolState?: StakePool;
   }): Promise<DecimalUnits> {
-    let userPosition = lastUserPositionState;
-    if (!userPosition) {
-      userPosition = await this.queryUserPosition(userAddress);
+    let userPositionState = userPosition;
+    if (!userPositionState) {
+      userPositionState = await this.queryUserPosition(userAddress);
     }
-    const accRewardPerShare = await this.calcAccRewardPerShare();
+    const accRewardPerShare = await this.calcAccRewardPerShare({
+      currentTimeUs,
+      lastPoolState,
+    });
+    const stakeAu = userPositionState.amountStaked
+      .toAtomicUnits(this.coinInfoStake.decimals)
+      .toNumber();
     return AU(
-      accRewardPerShare
-        .sub(userPosition.lastAccRewardPerShare)
-        .mul(
-          userPosition.amountStaked
-            .toAtomicUnits(this.coinInfoStake.decimals)
-            .toBN()
-        )
-        .div(REWARD_PER_SHARE_MUL)
+      Math.floor(
+        ((accRewardPerShare.toNumber() -
+          userPositionState.lastAccRewardPerShare.toNumber()) *
+          stakeAu) /
+          1e12
+      )
     ).toDecimalUnits(this.coinInfoReward.decimals);
   }
 
+  async createEvents(
+    query?: { start: BN } | { limit: BN }
```

Comment:
this looks off, i think you mean something like

```typescript
{
   start?: BN,
   limit?: BN
}
```
`x | y` means only one or the other but you'd want to specify both

Reply:
I just used the pattern from here: https://github.com/aux-exchange/aux-exchange/blame/a9113e95dbf4bb26fc338d815e0206d51f633de7/aptos/api/aux-ts/src/pool/constant-product/client.ts#LL453C10-L453C10

Doesn't actually look like this arg is ever used for constant product but I assumed there was a reason to have the arg.

Reply:
But yeah if this seems wrong I can just fix across the board. Would be helpful to understand what the meaning of these actually is though... to me `first` seems to map to `offset` but `start` and `limit` don't really seem the same.

Reply:
yeah, sorry i noticed that too after -- i didn't originally write that code

Reply:
alright. pagination isn't being used by any of the resolvers right now anyway so would prefer to take this as a follow-up since it was a pre-existing issue.